querydb/src/main/scala/io/joern/scanners/android/JavaScriptInterface.scala

```diff
@@ -0,0 +1,518 @@
+package io.joern.scanners.android
+
+import io.joern.scanners._
+import io.joern.console._
+import io.joern.dataflowengineoss.queryengine.EngineContext
+import io.joern.dataflowengineoss.semanticsloader.Semantics
+import io.joern.macros.QueryMacros._
+import io.shiftleft.semanticcpg.language._
+import io.joern.dataflowengineoss.language._
+
+object JavaScriptInterface extends QueryBundle {
+  implicit val engineContext: EngineContext = EngineContext(Semantics.empty)
+  implicit val resolver: ICallResolver      = NoResolve
+
+  // TODO: take into account network_security_config
+  // see: https://support.google.com/faqs/answer/9095419?hl=en
+  @q
+  def insecureLoadUrlToExec()(implicit engineContext: EngineContext): Query =
+    Query.make(
+      name = "insecure-load-url-to-exec",
+      author = Crew.claudiu,
+      title = "Data from an insecure url load reaches `Runtime.getRuntime.exec` via JavaScript bridge.",
+      description = "-",
+      score = 9,
+      withStrRep({ cpg =>
+        import overflowdb.traversal.Traversal
+        import io.joern.querydb.language.android.ManifestXmlTraversal
+        import io.shiftleft.codepropertygraph.generated.nodes.{Call, Identifier}
+        def webViews =
+          cpg.local.typeFullNameExact("android.webkit.WebView")
+        def webViewsWithJSEnabled =
+          webViews
+            .where(
+              _.referencingIdentifiers.inCall
+                .nameExact("getSettings")
+                .where(
+                  _.inCall
+                    .nameExact("setJavaScriptEnabled")
+                    .argument
+                    .isLiteral
+                    .codeExact("true")
+                )
+            )
+        def webViewsWithInsecureLoadUrlCalls =
+          webViewsWithJSEnabled.where(
+            _.referencingIdentifiers.inCall
+              .nameExact("loadUrl")
+              .filter { callNode =>
+                def httpLiterals =
+                  callNode.method.literal.filter(_.code.stripPrefix("\"").stripSuffix("\"").startsWith("http:"))
+                callNode.argument.reachableBy(httpLiterals).nonEmpty
+              }
+          )
+        val appUsesCleartextTraffic =
+          new ManifestXmlTraversal(cpg.configFile).usesCleartextTraffic.nonEmpty
+        def exposedJavaScriptInterfaceObjects =
+          if (appUsesCleartextTraffic)
+            webViewsWithInsecureLoadUrlCalls.referencingIdentifiers.inCall
+              .nameExact("addJavascriptInterface")
+              .argument(1)
+          else Traversal.empty
+        val exposedJavaScriptInterfaceObjectNames =
+          exposedJavaScriptInterfaceObjects.collect {
+            case expression: Identifier => expression.typeFullName
+            case expression: Call       => expression.typeFullName
```

Comment:
```suggestion
            case ident: Identifier => ident.typeFullName
            case call: Call        => call.typeFullName
```

since `Expression` is a node type, I think it's preferable to not use it as a variable name for a different type (even if it's a sub-type).

Reply:
yepp, nicer, thx